drivers/dac/dac_demo/dac_demo.c

```diff
@@ -173,9 +175,11 @@ int32_t dac_write_samples(void* dev, uint16_t* buff, uint32_t samples)
 		return -EINVAL;
 
 	for(uint32_t i = 0; i < samples; i++)
-		while (get_next_ch_idx(desc->active_ch, ch, &ch))
-			desc->loopback_buffers[ch]
-			[i % desc->loopback_buffer_len] = buff[k++];
+		while (get_next_ch_idx(desc->active_ch, ch, &ch)) {
+			uint16_t* ch_buffer = (uint16_t*)(desc->loopback_buffers+
+							  (ch*desc->loopback_buffer_len*sizeof(uint16_t)/sizeof(ch_buffer)));
+			ch_buffer[i % desc->loopback_buffer_len] = buff[k++];
+		}
```

Comment:
nitpick: is astyle applied here? We shouldn't have lines longer than 80 chars, it ends up looking odd.